src/Bicep.Core/TypeSystem/Types/IUnboundResourceDerivedType.cs

```diff
@@ -13,13 +14,11 @@ namespace Bicep.Core.TypeSystem.Types;
 /// </summary>
 public interface IUnboundResourceDerivedType
```

Comment:
[nit] The 'unbound' terminology has been confusing me (since we already have a separate unrelated concept of binding), and it seems like it's being used to refer specifically just to resource derived types. Have you considered calling this just `IResourceDerivedType`?

Reply:
Maybe `IUnresolvedResourceDerivedType` is better? At this point, the compiler hasn't yet looked up the type as defined by the provider.